Add status box in the Product Feed page

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Closes #716 

- Add card layout to the statistics block and move the help popover into card header
- Implement the \<StatusBox\> and add it into card footer

### ❗ Additional note

With the enhancement of prefetch product feed data made by #629, if I try to re-fetch the product statistics data for updating the "Sync with Google" status, the whole product feed page would be replaced by displaying a loading spinner during the data loading. So I didn't implement the part of status auto-update in this PR.

### Screenshots:

#### 📷 . Has unsolved issues and sync in progress

![2021-06-03 19 23 04](https://user-images.githubusercontent.com/17420811/120639257-bdebe980-c4a3-11eb-8880-6f972f89314d.png)

#### 📷 . Resolved all issues and synced all products

![2021-06-03 19 21 39](https://user-images.githubusercontent.com/17420811/120639276-c0e6da00-c4a3-11eb-8090-44e19957b9b9.png)

### Detailed test instructions:

Mock the `mc/product-statistics` API with the following snippet in `google-listings-and-ads.php`
```php
add_filter(
	'gla_prepared_response_mc-product-statistics',
	function( $response ) {
		return [
			'timestamp' => 1622718951,
			'scheduled_sync' => 1,
			'statistics' => [
				'active' => 100,
				'expiring' => 200,
				'pending' => 300,
				'disapproved' => 400,
				'not_synced' => 500
			]
		];
	},
);
```

Mock the `mc/ssues` API with the following snippet in `google-listings-and-ads.php`
```php
add_filter(
	'gla_prepared_response_mc-issues',
	function( $response ) {
		return [
			'total' => 777,
			'page' => 1,
			'issues' => []
		];
	},
);
```

1. Go to the Product Feed page
2. Change mock data to see if the render logics and visual are correct

### Changelog Note:

> Add status box in the Product Feed page

eason9487 committed:
Extract PER_PAGE of issue table to constants

eason9487 committed:
Implement the StatusBox for the product feed page

eason9487 committed:
Add card layout and move the help popover for ProductStatistics

eason9487 committed:
Add StatusBox to ProductStatistics

j111q reviewed (approved):
Cool, thanks for the clarifications everyone!  👏

eason9487 committed:
Handle statut box layout for smaller screens

ecgan reviewed (approved):
Tested working OK, LGTM. Just left some small nit questions and comments.

ecgan reviewed (commented):
Add another small nit comment on `useMemo`.

eason9487 committed:
Take a more robust way to check invalid value for getUnsolvedStatusText

Address https://github.com/woocommerce/google-listings-and-ads/pull/726/files#r646428954

eason9487 committed:
Remove useMemo from the SyncStatus component

Address https://github.com/woocommerce/google-listings-and-ads/pull/726/files#r646486917
